Add npm1012 support

kopuu1 opened the pull request.

kopuu1 committed:
Add npm1012 device

kopuu1 committed:
Add npm1012 device

kopuu1 committed:
npm1012: Add minimal set of unit test

kopuu1 committed:
npm1012: Add minimal set of unit test

kopuu1 committed:
npm1012/tests/exportImport.test.ts: Add file

kopuu1 committed:
npm1012/tests/exportImport.test.ts: Add file

kopuu1 committed:
npm1012: Copy over charger module from 1300

kopuu1 committed:
npm1012: Copy over charger module from 1300

kopuu1 committed:
npm1012: Add the charger module to the device

kopuu1 committed:
npm1012: Add the charger module to the device

kopuu1 committed:
npm/types/charger: Add vTricklefast and optional iTrickle for 1012

kopuu1 committed:
npm/types/charger: Add vTricklefast and optional iTrickle for 1012

kopuu1 committed:
npm1012/charger: Add iTrickle command support

kopuu1 committed:
npm1012/charger: Add iTrickle command support

kopuu1 committed:
PowerCard: Add optional iTrickle dropdown

kopuu1 committed:
PowerCard: Add optional iTrickle dropdown

kylebonnici reviewed (changes requested):
Very good start 🥇 

 nice work :)

kylebonnici commented on src/features/pmicControl/npm/types.ts:
```diff
@@ -520,6 +527,7 @@ export abstract class ChargerModuleSetBase {
     abstract vTerm(value: number): Promise<void>;
     abstract iChg(value: number): Promise<void>;
     abstract enabled(value: boolean): Promise<void>;
+    iTrickle?(value: ITrickle): Promise<void>;
```
I see the issue so then lets remove `abstract` form optional methods i.e. `abstract batLim?(value: number): Promise<void>;`

kopuu1 committed:
npm/types/charger: Remove abstract from optional methods

kopuu1 committed:
npm/types/charger: Remove abstract from optional methods

kopuu1 committed:
npm/types/charger: Change current range type in ChargerModuleRanges to number array

kopuu1 committed:
npm/types/charger: Change current range type in ChargerModuleRanges to number array

kopuu1 committed:
npm1012/charger: Modify VTerm range

kopuu1 committed:
npm1012/charger: Modify VTerm range

kopuu1 committed:
npm/pmicHelpers: Update 1012 USB id's

kopuu1 committed:
npm/pmicHelpers: Update 1012 USB id's

kopuu1 committed:
npm1012/tests/helpers: Fix number of GPIO's

kopuu1 committed:
npm1012/tests/helpers: Fix number of GPIO's

kopuu1 committed:
npm1012: Disable generateOverlay() for now

kopuu1 committed:
npm1012: Disable generateOverlay() for now

kopuu1 committed:
deviceSetups: Remove useless comment from npm1012DeviceSetup

kopuu1 committed:
deviceSetups: Remove useless comment from npm1012DeviceSetup

kylebonnici commented on src/components/Battery/BatteryStatus.tsx:
```diff
@@ -79,10 +79,7 @@ export default ({ disabled }: { disabled: boolean }) => {
-                                  npmDevice?.chargerModule?.ranges.current
-                                      .decimals ?? 0,
-                              )} mA`
+                              ).toFixed(2)} mA`
```
What is the motivation behind this change?

kylebonnici commented on src/features/pmicControl/npm/npm1300/charger/index.ts:
```diff
@@ -141,6 +136,17 @@ export default class Module implements ChargerModuleBase {
+            },
+        ]).map(v => Number(v));
+    }
+
```
Revert back not needed

kopuu1 committed:
npm1012/charger: Modify current range

kopuu1 committed:
npm1012/charger: Modify current range

kopuu1 committed:
utils/helpers: Add RangeOrNumberArray type with helper functions

kopuu1 committed:
utils/helpers: Add RangeOrNumberArray type with helper functions

kopuu1 committed:
npm/types/charger: Change current range type to RangeOrNumberArray

kopuu1 committed:
npm/types/charger: Change current range type to RangeOrNumberArray

kopuu1 committed:
Revert "npm/types/charger: Change current range type in ChargerModuleRanges to number array"

This reverts commit 618ad208199bd08f235814e2d9db6e3b0fc801cf.

kopuu1 committed:
Revert "npm/types/charger: Change current range type in ChargerModuleRanges to number array"

This reverts commit 618ad208199bd08f235814e2d9db6e3b0fc801cf.

kylebonnici commented on src/utils/helpers.ts:
```diff
@@ -47,3 +47,29 @@ export const getRange = (ranges: RangeType[]): number[] => {
+): number =>
+    Array.isArray(values)
+        ? Math.min(...values)
+        : Math.min(...getRange([values]));
```
```suggestion
    Array.isArray(values)
        ? Math.min(...values)
        : values.min
```

kopuu1 committed:
utils/helpers: Add few improvements

kopuu1 committed:
utils/helpers: Add few improvements

kopuu1 committed:
npm/types/charger: Add optional enableWeakBatteryCharging

kopuu1 committed:
npm/types/charger: Add optional enableWeakBatteryCharging

kopuu1 committed:
npm1012/charger: Add support for enabledWeakBatteryCharging, remove batLim

kopuu1 committed:
npm1012/charger: Add support for enabledWeakBatteryCharging, remove batLim

kopuu1 committed:
PowerCard: Add optional enableWeakBatteryCharging

kopuu1 committed:
PowerCard: Add optional enableWeakBatteryCharging

kopuu1 committed:
npm/types/charger: Add optional vWeak

kopuu1 committed:
npm/types/charger: Add optional vWeak

kopuu1 committed:
npm1012/charger: Add support for vWeak

kopuu1 committed:
npm1012/charger: Add support for vWeak

kopuu1 committed:
PowerCard: Add optional vWeak

kopuu1 committed:
PowerCard: Add optional vWeak

kopuu1 committed:
npm1012/charger: Add missing callbacks

kopuu1 committed:
npm1012/charger: Add missing callbacks

kylebonnici reviewed (commented):
look good now 🥇

kopuu1 committed:
npm/types/charger: Add jeita labels

kopuu1 committed:
npm/types/charger: Add jeita labels

kopuu1 committed:
Charger/Jeita.tsx: Add external label control

kopuu1 committed:
Charger/Jeita.tsx: Add external label control

kopuu1 committed:
Charger/Charger.tsx: Increase MasonryLayout minWidth

- Width needs to be higher after an update to the Jeita card

kopuu1 committed:
Charger/Charger.tsx: Increase MasonryLayout minWidth

- Width needs to be higher after an update to the Jeita card

kopuu1 committed:
npm1012/charger: Add default jeita labels

kopuu1 committed:
npm1012/charger: Add default jeita labels

kopuu1 committed:
npm130x/charger: Add default jeita labels

kopuu1 committed:
npm130x/charger: Add default jeita labels

kopuu1 committed:
npm2100/charger: Add default jeita labels

kopuu1 committed:
npm2100/charger: Add default jeita labels

kopuu1 committed:
npm/types/charger: Require iChg input parameter for getting values.iTrickle

kopuu1 committed:
npm/types/charger: Require iChg input parameter for getting values.iTrickle

kopuu1 committed:
npm1012/charger: Add getITrickleValues() with iChg as input parameter

kopuu1 committed:
npm1012/charger: Add getITrickleValues() with iChg as input parameter

kopuu1 committed:
charger/PowerCard: Get iTrickle values using iChg as input parameter

kopuu1 committed:
charger/PowerCard: Get iTrickle values using iChg as input parameter

kopuu1 committed:
npm1012/charger: Add iChgCool, iChgWarm, vTermCool, vTermWarm

- Add commands and callbacks

kopuu1 committed:
npm1012/charger: Add iChgCool, iChgWarm, vTermCool, vTermWarm

- Add commands and callbacks

kopuu1 committed:
npm/types: Add iChgCool, iChgWarm, vTermCool, vTermWarm

- Add optional types with get/set commands

kopuu1 committed:
npm/types: Add iChgCool, iChgWarm, vTermCool, vTermWarm

- Add optional types with get/set commands

kopuu1 committed:
npm/types: Add enableAdvancedChargingProfile and enableNtcMonitoring

- Add optional types with set commands

kopuu1 committed:
npm/types: Add enableAdvancedChargingProfile and enableNtcMonitoring

- Add optional types with set commands

kopuu1 committed:
npm1012/charger: Add enableAdvancedChargingProfile and enableNtcMonitoring

- Use enableAdvancedChargingProfile to change jeita ui labels

kopuu1 committed:
npm1012/charger: Add enableAdvancedChargingProfile and enableNtcMonitoring

- Use enableAdvancedChargingProfile to change jeita ui labels

kopuu1 committed:
Charger/Jeita.tsx: Add support for advanced charging profile

kopuu1 committed:
Charger/Jeita.tsx: Add support for advanced charging profile

kopuu1 committed:
Charger/Jeita.tsx: Add option to disable number input for temperature

kopuu1 committed:
Charger/Jeita.tsx: Add option to disable number input for temperature

kopuu1 committed:
npm/types: Add types and methods for battery life and safety card

kopuu1 committed:
npm/types: Add types and methods for battery life and safety card

kopuu1 committed:
npm1012/charger: Add functionality for battery life and safety card

kopuu1 committed:
npm1012/charger: Add functionality for battery life and safety card

kopuu1 committed:
Charger: Add battery life and safety card

kopuu1 committed:
Charger: Add battery life and safety card

kopuu1 committed:
npm1012/charger: Move iTrickle values get method

kopuu1 committed:
npm1012/charger: Move iTrickle values get method

kopuu1 committed:
npm/types/charger: Add iChg as an optional parameter for charger.values.iTerm()

kopuu1 committed:
npm/types/charger: Add iChg as an optional parameter for charger.values.iTerm()

kopuu1 committed:
npm1012/charger: Fix iTerm values in get values()

kopuu1 committed:
npm1012/charger: Fix iTerm values in get values()

kopuu1 committed:
npm130x/charger: Change get values.iTerm return to have iChg parameter

- iChg parameter not used as it is not needed for getting the iTerm values on
  npm130x devices

kopuu1 committed:
npm130x/charger: Change get values.iTerm return to have iChg parameter

- iChg parameter not used as it is not needed for getting the iTerm values on
  npm130x devices

kopuu1 committed:
PowerCard.tsx: Use iChg as a parameter for getting iTerm values

kopuu1 committed:
PowerCard.tsx: Use iChg as a parameter for getting iTerm values

kopuu1 committed:
ConfigurationDialog.tsx: Use iChg as a parameter for getting iTerm values

kopuu1 committed:
ConfigurationDialog.tsx: Use iChg as a parameter for getting iTerm values

kopuu1 committed:
npm1012/charger/getters.ts: Add missing vWeak() call from all()

kopuu1 committed:
npm1012/charger/getters.ts: Add missing vWeak() call from all()

kopuu1 committed:
PowerCard.tsx: Remove 'Charging Below Vbatlow' toggle

- Moved to 'Battery Life and Safety' card

kopuu1 committed:
PowerCard.tsx: Remove 'Charging Below Vbatlow' toggle

- Moved to 'Battery Life and Safety' card

kylebonnici reviewed (dismissed):
Looks good to me 


We still have one more case to consider 

lets assume that iTerm is set to 1.56% and then iChg is changed to < 8ma .... will FW Throw... do we have to first change the Iterm etc... or will FW choose a suitable alternative? 

This can be taken care of in some other PR IMO hence I am approving this PR

Nice work!!

kopuu1 committed:
npm1012/charger/setters: Remove unneeded undefined check for functions

kopuu1 committed:
npm1012/charger/setters: Remove unneeded undefined check for functions

kopuu1 committed:
npm/types/charger: Make ntcBeta and ntcThermistor optional

kopuu1 committed:
npm/types/charger: Make ntcBeta and ntcThermistor optional

kopuu1 committed:
Charger/Jeita.tsx: Handle nTCBeta and nTCThermistor as optional

kopuu1 committed:
Charger/Jeita.tsx: Handle nTCBeta and nTCThermistor as optional

kopuu1 committed:
ChecklistDialog.tsx: Handle set.nTCThermistor() as optional

kopuu1 committed:
ChecklistDialog.tsx: Handle set.nTCThermistor() as optional

kopuu1 committed:
npm130x/charger: Handle methods for nTC Beta and Thermistor as optional

kopuu1 committed:
npm130x/charger: Handle methods for nTC Beta and Thermistor as optional

kopuu1 committed:
npm1012/charger: Remove unused code for ntc beta and thermistor

kopuu1 committed:
npm1012/charger: Remove unused code for ntc beta and thermistor

kopuu1 committed:
npm1012/charger: Add missing commands for advanced charging profile

kopuu1 committed:
npm1012/charger: Add missing commands for advanced charging profile

kopuu1 committed:
npm/types/charger: Add missing get commands for advanced charging profile

kopuu1 committed:
npm/types/charger: Add missing get commands for advanced charging profile

kopuu1 committed:
npm1012/charger: Remove unneeded disabling of charging before sending commands

kopuu1 committed:
npm1012/charger: Remove unneeded disabling of charging before sending commands

kopuu1 committed:
npm1012/charger/setters: Remove unneeded undefined check for functions

kopuu1 committed:
npm1012/charger/setters: Remove unneeded undefined check for functions

kopuu1 committed:
npm1012/charger: Keep type as RangeType for values

kopuu1 committed:
npm1012/charger: Keep type as Range for values

kopuu1 committed:
npm/types/charger: Make tChgStop optional, add tChgReduce

kopuu1 committed:
npm/types/charger: Make tChgStop optional, add tChgReduce

kopuu1 committed:
npm130x/charger: Handle tChgStop as optional

kopuu1 committed:
npm130x/charger: Handle tChgStop as optional

kopuu1 committed:
npm1012: Add buck module

kopuu1 committed:
npm1012: Add buck module

kopuu1 committed:
pmic1012Device.tsx: Add BuckModule to peripherals

kopuu1 committed:
pmic1012Device.tsx: Add BuckModule to peripherals

kopuu1 committed:
npm1012/charger: Add methods for tChgReduce, remove tChgStop

kopuu1 committed:
npm1012/charger: Add methods for tChgReduce, remove tChgStop

kopuu1 committed:
Charger/ThermalRegulation.tsx: Add tChgReduce

- Show number inputs and slider for tChgResume and tChgStop when tChgStop is
  defined, and for tChgResume and tChgReduce when tChgReduce is
  defined.

kopuu1 committed:
Charger/ThermalRegulation.tsx: Add tChgReduce

- Show number inputs and slider for tChgResume and tChgStop when tChgStop is
  defined, and for tChgResume and tChgReduce when tChgReduce is
  defined.

kopuu1 committed:
npm1012/charger: Add callback for tChgReduce

kopuu1 committed:
npm1012/charger: Add callback for tChgReduce

kopuu1 committed:
npm/types/buck: Add new types and properties for npm1012

kopuu1 committed:
npm/types/buck: Add new types and properties for npm1012

kopuu1 committed:
npm130x/buck: Handle updated properties for Buck and BuckModule types

kopuu1 committed:
npm130x/buck: Handle updated properties for Buck and BuckModule types

kopuu1 committed:
BuckCard.tsx: Add new control options for npm1012

kopuu1 committed:
BuckCard.tsx: Add new control options for npm1012

kopuu1 committed:
pmic1012Device.tsx: Add LdoModule

kopuu1 committed:
pmic1012Device.tsx: Add LdoModule

kopuu1 committed:
npm1012: Add ldo module

kopuu1 committed:
npm1012: Add ldo module

kopuu1 committed:
npm/types/ldo: Update properties for npm1012

kopuu1 committed:
npm/types/ldo: Update properties for npm1012

kopuu1 committed:
LDOCard.tsx: Update for npm1012

kopuu1 committed:
LDOCard.tsx: Update for npm1012

kopuu1 committed:
npm1300/gpio/types: Add numGPIOs constant

kopuu1 committed:
npm1300/gpio/types: Add numGPIOs constant

kopuu1 committed:
npm1300/pmic1300Device.tsx: Use numGPIOs constant

kopuu1 committed:
npm1300/pmic1300Device.tsx: Use numGPIOs constant

kopuu1 committed:
Regulators.tsx: Delete unneeded numberOfGPIOs parameter for BuckCard

kopuu1 committed:
Regulators.tsx: Delete unneeded numberOfGPIOs parameter for BuckCard

kopuu1 committed:
DashboardControl.tsx: Delete unneeded numberOfGPIOs parameter for BuckCard

kopuu1 committed:
DashboardControl.tsx: Delete unneeded numberOfGPIOs parameter for BuckCard

kopuu1 committed:
useNpmDevice.tsx: Fix hiding the "System Features" tab if not a supported device

kopuu1 committed:
useNpmDevice.tsx: Fix hiding the "System Features" tab if not a supported device

kopuu1 committed:
npm1012/charger: Update shell commands

kopuu1 committed:
npm1012/charger: Update shell commands

kopuu1 committed:
DeviceSelector.tsx: Add support for npm1012

kopuu1 committed:
DeviceSelector.tsx: Add support for npm1012

kopuu1 committed:
pmic1012Device.tsx: Fix initial FW version

kopuu1 committed:
pmic1012Device.tsx: Fix initial FW version

kopuu1 committed:
npm/types/charger: Make vTermR optional

kopuu1 committed:
npm/types/charger: Make vTermR optional

kopuu1 committed:
Jeita.tsx: Handle vTermR as optional

kopuu1 committed:
Jeita.tsx: Handle vTermR as optional

kopuu1 committed:
npm130x/charger: Handle vTermR as optional

kopuu1 committed:
npm130x/charger: Handle vTermR as optional

kopuu1 committed:
npm1300/overlay: Handle vTermR as optional

kopuu1 committed:
npm1300/overlay: Handle vTermR as optional

kopuu1 committed:
npm130x/ldo: Sync with updated Ldo and LdoModule types

kopuu1 committed:
npm130x/ldo: Sync with updated Ldo and LdoModule types

kopuu1 committed:
npm2100/ldo: Sync with updated Ldo and LdoModule types

kopuu1 committed:
npm2100/ldo: Sync with updated Ldo and LdoModule types

kopuu1 committed:
Merge branch 'update_thermal_regulation_card' into add_npm1012_support

kopuu1 committed:
Merge branch 'add_battery_life_and_safety_card' into add_npm1012_support

kopuu1 committed:
BatteryLifeAndSafety.tsx: Add toggle for enableBatteryDischargeCurrentLimit

kopuu1 committed:
BatteryLifeAndSafety.tsx: Add toggle for enableBatteryDischargeCurrentLimit

kopuu1 committed:
npm1012/charger: Add tests

kopuu1 committed:
npm1012/charger: Add tests

kopuu1 committed:
package.json: Update nrfutil device version

kopuu1 committed:
PowerCard.tsx: Move iTrickle, vWeak and weakBatteryCharging under default UI

kopuu1 committed:
PowerCard.tsx: Move iTrickle, vWeak and weakBatteryCharging under default UI

kopuu1 committed:
pmic1012Device.tsx: Override startAdcSample() in BaseNpmDevice

kopuu1 committed:
pmic1012Device.tsx: Override startAdcSample() in BaseNpmDevice

kopuu1 committed:
npm1012/tests: Update requestUpdates test

kopuu1 committed:
npm1012/tests: Update requestUpdates test

kopuu1 committed:
Regulators.tsx: Delete unneeded parameters to LDOCard

kopuu1 committed:
Regulators.tsx: Delete unneeded cardLabel parameter to LDOCard

kopuu1 committed:
DashboardControl.tsx: Delete unneeded cardLabel parameter to LDOCard

kopuu1 committed:
DashboardControl.tsx: Delete unneeded parameters to LDOCard

kopuu1 committed:
fw: Add initial npm1012 firmware version 0.0.1+0

kopuu1 committed:
fw: Add initial npm1012 firmware version 0.0.1+0

kopuu1 committed:
npm/types/charger: Delete unneeded iChg parameter to get iTerm and iTrickle values

kopuu1 committed:
npm/types/charger: Delete unneeded iChg parameter to get iTerm and iTrickle values

kopuu1 committed:
PowerCard.tsx: Delete unneeded iChg parameter to get iTerm and iTrickle values

kopuu1 committed:
PowerCard.tsx: Delete unneeded iChg parameter to get iTerm and iTrickle values

kopuu1 committed:
ConfigurationDialog.tsx: Delete unneeded iChg parameter to get iTerm values

kopuu1 committed:
ConfigurationDialog.tsx: Delete unneeded iChg parameter to get iTerm values

kopuu1 committed:
npm1012/charger: Delete iChg parameter to get iTerm and iTrickle values

kopuu1 committed:
npm1012/charger: Delete iChg parameter to get iTerm and iTrickle values

kopuu1 committed:
npm130x/charger: Sync with updated ChargerModuleValues type

kopuu1 committed:
npm130x/charger: Sync with updated ChargerModuleValues type

kopuu1 committed:
npm/types: Add LedModule

kopuu1 committed:
npm/types: Add LedModule

kopuu1 committed:
src/components/GPIOs/LEDs.tsx: Delete file

kopuu1 committed:
src/components/GPIOs/LEDs.tsx: Delete file

kopuu1 committed:
src/components/GPIOs/LED.tsx: Add file

kopuu1 committed:
src/components/GPIOs/LED.tsx: Add file

kopuu1 committed:
GPIOs.tsx: Use ledModule

kopuu1 committed:
GPIOs.tsx: Use ledModule

kopuu1 committed:
npm/basePmicDevice.ts: Add ledModule

kopuu1 committed:
npm/basePmicDevice.ts: Add ledModule

kopuu1 committed:
useNpmDevice.tsx: Use ledModule

kopuu1 committed:
useNpmDevice.tsx: Use ledModule

kopuu1 committed:
pmic1012Device.tsx: Add LedModule

kopuu1 committed:
pmic1012Device.tsx: Add LedModule

kopuu1 committed:
pmic1300Device.tsx: Use LedModule

kopuu1 committed:
pmic1300Device.tsx: Use LedModule

kopuu1 committed:
pmic2100Device.tsx: Delete unused noOfLEDs from NpmPeripherals

kopuu1 committed:
pmic2100Device.tsx: Delete unused noOfLEDs from NpmPeripherals

kopuu1 committed:
npm1012: Add LedModule

p#	new file:   src/features/pmicControl/npm/npm1012/led/callbacks.test.ts

kopuu1 committed:
npm1012: Add LedModule

p#	new file:   src/features/pmicControl/npm/npm1012/led/callbacks.test.ts

kopuu1 committed:
npm130x: Add led module

kopuu1 committed:
npm130x: Add led module

kopuu1 committed:
src/index.tsx: Add LEDs to GPIOs tab title

kopuu1 committed:
src/index.tsx: Add LEDs to GPIOs tab title

kopuu1 committed:
Merge branch 'add_ldo_module_for_npm1012' into add_regulators_for_npm1012

kopuu1 committed:
Merge branch 'add_regulators_for_npm1012' into add_npm1012_support

kopuu1 committed:
fw: Update npm1012 firmware version to 0.0.2+0

kopuu1 committed:
fw: Update npm1012 firmware version to 0.0.2+0

kopuu1 committed:
npm1012/buck: Edit

- Add confirmation dialogs when disabling buck or when vout is set below 1.62 V
- Add units to shell commands
- Change mode to software when vout is changed
- Change onOffControl to software or vset when mode is changed
- Update tests

kopuu1 committed:
npm1012/buck: Edit

- Add confirmation dialogs when disabling buck or when vout is set below 1.62 V
- Add units to shell commands
- Change mode to software when vout is changed
- Change onOffControl to software or vset when mode is changed
- Update tests

kopuu1 committed:
npm1012/charger: Edit

- Remove get.state() when enable is toggled
- Add initial support for charger state command
- Update tests

kopuu1 committed:
npm1012/charger: Edit

- Remove get.state() when enable is toggled
- Add initial support for charger state command
- Update tests

kopuu1 committed:
npm1012/overlay: Delete overlay for LED until format is known

kopuu1 committed:
npm1012/overlay: Delete overlay for LED until format is known

kopuu1 committed:
npm1012/tests/helpers: Change number of leds

kopuu1 committed:
npm1012/tests/helpers: Change number of leds

kopuu1 committed:
npm130x/tests: Delete led mode tests

- Moved to led module

kopuu1 committed:
npm130x/tests: Delete led mode tests

- Moved to led module

kopuu1 committed:
npm1300/overlay: Check for undefined mode in generateLEDs()

kopuu1 committed:
npm1300/overlay: Check for undefined mode in generateLEDs()

kopuu1 committed:
npm130x/tests/exportImport: Add missing cardLabel to initLed

kopuu1 committed:
npm130x/tests/exportImport: Add missing cardLabel to initLed

kopuu1 committed:
npm2100/tests/exportImport: Add missing cardLabel to initLed

kopuu1 committed:
npm2100/tests/exportImport: Add missing cardLabel to initLed

kopuu1 committed:
npm2100/tests: Delete unneeded led mode tests

- No led control in the npm2100, only gpio

kopuu1 committed:
npm2100/tests: Delete unneeded led mode tests

- No led control in the npm2100, only gpio

kopuu1 committed:
pmicControlSlice.ts: Fix indexing in updateLEDs()

kopuu1 committed:
pmicControlSlice.ts: Fix indexing in updateLEDs()

kopuu1 committed:
npm/types: Add GPIOLEDDrv

kopuu1 committed:
npm/types: Add GPIOLEDDrv

kopuu1 committed:
pmicControl/npm/basePmicDevice.ts: Add gpioLedDrvModule

kopuu1 committed:
pmicControl/npm/basePmicDevice.ts: Add gpioLedDrvModule

kopuu1 committed:
pmic1012Device.tsx: Add gpioLedDrvs

kopuu1 committed:
pmic1012Device.tsx: Add gpioLedDrvs

kopuu1 committed:
npm1012: Add gpioleddrv

kopuu1 committed:
npm1012: Add gpioleddrv

kopuu1 committed:
pmic1012Device.tsx: Enable startAdcSample()

kopuu1 committed:
pmic1012Device.tsx: Enable startAdcSample()

kopuu1 committed:
pmic1012Device.tsx: Add FuelGaugeModule

kopuu1 committed:
pmic1012Device.tsx: Add FuelGaugeModule

kopuu1 committed:
npm1012: Add fuelGauge module

kopuu1 committed:
npm1012: Add fuelGauge module

kopuu1 committed:
pmic1012Device.tsx: Add BatteryProfiler

kopuu1 committed:
pmic1012Device.tsx: Add BatteryProfiler

kopuu1 committed:
npm1012: Add batteryProfiler

kopuu1 committed:
npm1012: Add batteryProfiler

kopuu1 committed:
pmic1012Device.tsx: Add UsbCurrentLimiterModule

kopuu1 committed:
pmic1012Device.tsx: Add UsbCurrentLimiterModule

kopuu1 committed:
npm1012: Add universalSerialBusCurrentLimiter

kopuu1 committed:
npm1012: Add universalSerialBusCurrentLimiter

kopuu1 committed:
GPIOs/GPIOLEDDrv.tsx: Add file

kopuu1 committed:
GPIOs/GPIOLEDDrv.tsx: Add file

kopuu1 committed:
pmicControlSlice.ts: Add GpioLedDrv

kopuu1 committed:
pmicControlSlice.ts: Add GpioLedDrv

kopuu1 committed:
GPIOs/GPIOs.tsx: Add GPIOLEDDrv

kopuu1 committed:
GPIOs/GPIOs.tsx: Add GPIOLEDDrv

kopuu1 committed:
useNpmDevice.tsx: Use set and update GPIOLEDDrv

kopuu1 committed:
useNpmDevice.tsx: Use set and update GPIOLEDDrv

kopuu1 committed:
npm/types/ldo: Add optional mode and vOutSel parameters to onOffControl values

kopuu1 committed:
npm/types/ldo: Add optional mode and vOutSel parameters to onOffControl values

kopuu1 committed:
LDOCard.tsx: Pass mode and vOutSel as parameters to values.onOffControl()

kopuu1 committed:
LDOCard.tsx: Pass mode and vOutSel as parameters to values.onOffControl()

kopuu1 committed:
npm1300/ldo/index: Update type for onOffControl values

kopuu1 committed:
npm1300/ldo/index: Update type for onOffControl values

kopuu1 committed:
npm1012/ldo: Edit

- Add mode and vOutSel parameters to values.onOffControl
- Disabling some of the functionality for Load Switch 2
- Change vOutSel and onOffControl to software when enable is toggled
- Change vOutSel to software when vOut is changed
- Update tests

kopuu1 committed:
npm1012/ldo: Edit

- Add mode and vOutSel parameters to values.onOffControl
- Disabling some of the functionality for Load Switch 2
- Change vOutSel and onOffControl to software when enable is toggled
- Change vOutSel to software when vOut is changed
- Update tests

kopuu1 committed:
resources/batteryModels: Copy npm130x models for npm1012

kopuu1 committed:
resources/batteryModels: Copy npm130x models for npm1012

kopuu1 committed:
pmic1012Device.tsx: Fix fuel gauge log event parsing

kopuu1 committed:
pmic1012Device.tsx: Fix fuel gauge log event parsing

kopuu1 committed:
npm/tests/helpers: Add mockOnGpioLedDrvUpdate

kopuu1 committed:
npm/tests/helpers: Add mockOnGpioLedDrvUpdate

kopuu1 committed:
npm1012/tests/helpers: Add GPIOLEDDRVS constant

kopuu1 committed:
npm1012/tests/helpers: Add GPIOLEDDRVS constant

kopuu1 committed:
fw: Update npm1012 firmware version to 0.0.3+0

kopuu1 committed:
fw: Update npm1012 firmware version to 0.0.3+0

kopuu1 committed:
npm1012/tests: Update logging and requestUpdates tests

kopuu1 committed:
npm1012/tests: Update logging and requestUpdates tests

kopuu1 committed:
npm1012: Add onBoardLoad module

kopuu1 committed:
npm1012: Add onBoardLoad module

kopuu1 committed:
pmic1012Device.tsx: Add onBoardLoad module

kopuu1 committed:
pmic1012Device.tsx: Add onBoardLoad module

kopuu1 committed:
Merge branch 'npm1012_add_gpios_and_leds' into add_npm1012_support

kopuu1 committed:
fw: Update npm1012 firmware version to 0.0.4+0

kopuu1 committed:
fw: Update npm1012 firmware version to 0.0.4+0

kopuu1 committed:
pmic1012Device.tsx: Disable battery profiling for now

kopuu1 committed:
pmic1012Device.tsx: Disable battery profiling for now

kopuu1 committed:
fw: Update npm1012 firmware version to 0.0.5+0

kopuu1 committed:
fw: Update npm1012 firmware version to 0.0.5+0

kopuu1 committed:
LDOCard.tsx: Sync onOffControl value when dropdown item is not available

kopuu1 committed:
LDOCard.tsx: Sync onOffControl value when dropdown item is not available

kopuu1 committed:
pmic1012Device.tsx: Request charger state update on adc logging event

kopuu1 committed:
pmic1012Device.tsx: Request charger state update on adc logging event

kopuu1 committed:
useNpmDevice.tsx: Poll buck.enable only with npm130x

kopuu1 committed:
useNpmDevice.tsx: Poll buck.enable only with npm130x

kopuu1 committed:
npm1012/gpioleddrv/setters: Set polarity to ACTIVELOW when GPIO is configured as an LED

kopuu1 committed:
npm1012/gpioleddrv/setters: Set polarity to ACTIVELOW when GPIO is configured as an LED

kopuu1 committed:
fw: Update npm1012 firmware version to 0.0.6+0

kopuu1 committed:
fw: Update npm1012 firmware version to 0.0.6+0

kopuu1 committed:
npm1012/universalSerialBusCurrentLimiter/callbacks: Handle VBUS irq log event

kopuu1 committed:
npm1012/universalSerialBusCurrentLimiter/callbacks: Handle VBUS irq log event

kopuu1 committed:
pmic1012Device.tsx: Edit log event handling

- Handle only known events

kopuu1 committed:
pmic1012Device.tsx: Edit log event handling

- Handle only known events

kopuu1 committed:
GPIOs/GPIOLEDDrv.tsx: Allow PWM duty cycle selection in blinking mode

kopuu1 committed:
npm/types/ldo: Add softStartCurrentDropdownDisabled

kopuu1 committed:
LDOCard.tsx: Add softStartCurrentDropdownDisabled

kopuu1 committed:
pmicControl/npmFactory: Pass pmic revision to npm1300 constructor

kopuu1 committed:
npm130x/ldo: Sync with LDO changes from main

- npm130x: Use softStartCurrentDropdownDisabled with older pmic revisions to
  grey out soft start current dropdown in LDO mode
- npm1300: Disable set.softStart due to no FW support
- npm1304: Implement LdoModule with pmicRevision check

kopuu1 committed:
npm1012/ldo/index.ts: Add softStart enable to export

kopuu1 committed:
npm1012/gpioleddrv: Edit pin configuration when using GPIO as LED output

kopuu1 committed:
npm130x/ldo/tests: Update tests

- Delete 'npmx soft_start enable' test as the shell command is unused in FW

kopuu1 committed:
npm130x/tests/exportImport: Update to ldo properties

kopuu1 committed:
fw: Update npm1012 firmware version to 0.1.0+0

kopuu1 committed:
npm1012/charger/callbacks.ts: Handle charger irq events

kopuu1 committed:
npm1012/universalSerialBusCurrentLimiter/callbacks.ts: Edit vbus irq event handling

kopuu1 committed:
npm/basePmicDevice.ts: Use partial event in onChargingStatusUpdate

kopuu1 committed:
pmicControlSlice.ts: Add updatePmicChargingState

kopuu1 committed:
pmicControl/npm/useNpmDevice.tsx: Use updatePmicChargingState

kopuu1 committed:
npm1012/pmic1012Device.tsx: Disable polling of charger state

kopuu1 committed:
npm1012/charger/callbacks.ts: Update charging state from the irq events

kopuu1 committed:
fw: Update npm1012 firmware version to 0.1.1+0

kopuu1 committed:
DeviceSelector.tsx: Remove npm1012 as a virtual device option

kopuu1 committed:
npm1012/led/index.ts: Add toLedExport()

kopuu1 committed:
pmic1012Device.tsx: Add buck and gpioLedDrv exports

kopuu1 committed:
fileActions.ts: Parse iTerm as float in upgradeToLatest()

kopuu1 committed:
npm1012/universalSerialBusCurrentLimiter: Add unit to vbusilim shell command

kopuu1 committed:
npm1012/universalSerialBusCurrentLimiter/tests: Update onlineSetter tests

kopuu1 committed:
fw: Update npm1012 firmware version to 0.2.0+0

kopuu1 committed:
npm1012/fuelGauge/setters.ts: Add batteryStatusCheckEnabled

kopuu1 committed:
npm1012/charger/callbacks.ts: Edit charging state parsing

kylebonnici commented on Changelog.md:
```diff
@@ -1,5 +1,9 @@
+### Added
+
+- Initial support for nPM1012.
+
```
Remove this for now AFAIK we should not be adding it to release notes.
